lib/rataextra-pipeline.ts

```diff
@@ -11,7 +11,7 @@ import { PolicyStatement, Effect } from 'aws-cdk-lib/aws-iam';
  */
 export class RataExtraPipelineStack extends Stack {
   constructor(scope: Construct) {
-    const config = getPipelineConfig();
+    const config = getPipelineConfig(scope);
```

Comment:
Presumably this scope can be undefined or something here? What if you call `super(scope)` in the line above and pass this to `getPipelineConfig`? That's how it is dome in rataextra-stack.ts. Give that a try, if that removes the need to handle undefined. If not, then no need to make changes. If it does, great. I'm already giving this approval, but do try that first before merging.

Reply:
I let `scope?: Construct` as optional in `getPipelineConfig()` cause cdk.ts doesn't have a class to pass `scope`

Reply:
Oh, I see! Thanks for the explanation. Feel free to merge this now.